Tests/XcodeGenKitTests/ProjectGeneratorTests.swift

```diff
@@ -299,6 +299,55 @@ func projectGeneratorTests() {
                 try project.expectFile(paths: ["Sources", "A", "B", "b.swift"], buildPhase: .sources)
             }
 
+            $0.it("generates source groups with excludes") {
+                let directories = """
+                Sources:
+                  A:
+                    - a.swift
+                    - B:
+                      - b.swift
+                  B:
+                    - b.swift
+                  C:
+                    - c.swift
+                    - c.m
+                    - c.h
+                  D:
+                    - d.h
+                    - d.m
+                  E:
+                    - e.jpg
+                    - e.h
+                    - e.m
+                    - F:
+                      - f.swift
+                """
+                try createDirectories(directories)
+
+                let excludes = [
+                    "B",
+                    "*.h",
+                    "f.swift"
```

Comment:
If we were using true glob patterns, `f.swift` wouldn't be removed, as it's in sub directory. Same goes for .h files that are in sub directories. That might be ok though, as this format is easier? What do you think?

Reply:
Is it currently possible to exclude certain files from a sub directory as we're not using the exact glob spec? If so can we add a case eg. `J/*.h`

Reply:
So the glob patterns we are using are recursive. I think that'd be more desirable for the ease of use. For example you'd like to exclude a certain type within a directory. Having said that, If you'd like to include a specific file you should provide the relative glob pattern, e.g. `E/F/f.swift`. This way every directory and subdirectory that has this pattern will be excluded.

Reply:
It is also possible to exclude file(s) from sub directory with a pattern. Will add the test case.

Reply:
Is there any other test cases you have in mind that you'd like me to add?

Reply:
What do you think of this? I have removed recursive patterns based on the reasoning above. 

Any other cases in mind?

```
$0.it("generates source groups with excludes") {
    let directories = """
    Sources:
        A:
        - a.swift
        - B:
            - b.swift
        B:
        - b.swift
        C:
        - c.swift
        - c.m
        - c.h
        D:
        - d.h
        - d.m
        E:
        - e.jpg
        - e.h
        - e.m
        - F:
            - f.swift
        G:
        H:
            - h.swift
    """
    try createDirectories(directories)

    let excludes = [
        "B",
        "C/*.h",
        "d.m",
        "E/F/*.swift",
        "G/H/"
    ]

    target.sources = [Source(path: "Sources", excludes: excludes)]
    spec.targets = [target]

    let project = try getPbxProj(spec)
    try project.expectFile(paths: ["Sources", "A", "a.swift"], buildPhase: .sources)
    try project.expectFile(paths: ["Sources", "C", "c.swift"], buildPhase: .sources)
    try project.expectFile(paths: ["Sources", "C", "c.m"], buildPhase: .sources)
    try project.expectFile(paths: ["Sources", "D", "d.h"])
    try project.expectFile(paths: ["Sources", "D", "d.m"], buildPhase: .sources)
    try project.expectFile(paths: ["Sources", "E", "e.jpg"], buildPhase: .resources)
    try project.expectFile(paths: ["Sources", "E", "e.m"], buildPhase: .sources)
    try project.expectFile(paths: ["Sources", "E", "e.h"])
    try project.expectFileMissing(paths: ["Sources/B", "b.swift"])
    try project.expectFileMissing(paths: ["Sources/C", "c.h"])
    try project.expectFileMissing(paths: ["Sources/E/F", "f.swift"])
    try project.expectFileMissing(paths: ["Sources/G/H", "h.swift"])
}

```

Reply:
I would like to have support in for proper recursive glob patterns.
Something like `**/*.h` which would remove all `.h` files. Is that possible with how you've implemented it? If so can we add a test case for it

Reply:
I think it should be possible. Will add it tomorrow :)